modules/server/src/main/scala/higherkindness/compendium/storage/FileStorage.scala

```diff
@@ -16,51 +16,65 @@
 
 package higherkindness.compendium.storage
 
-import java.io.{File, PrintWriter}
+import java.io.{File, FilenameFilter, PrintWriter}
 
 import cats.effect.Sync
 import cats.implicits._
-import higherkindness.compendium.core.refinements.ProtocolId
+import higherkindness.compendium.core.refinements.{ProtocolId, ProtocolVersion}
 import higherkindness.compendium.models.{FullProtocol, Protocol, ProtocolMetadata}
 import higherkindness.compendium.models.config.FileStorageConfig
 
 object FileStorage {
 
+  /**
+   * File pattern should match `{protocol identifier}_{version}.{extension}` where:
+   *
+   * - protocol identifier should comply with
+   * [[higherkindness.compendium.core.refinements.ProtocolId]] predicates
+   * - version should be a positive zero-leftpadded five digits version number like 00001
+   * - extension is `protocol`
+   */
+  private[storage] def buildFilename(id: ProtocolId, version: ProtocolVersion): String =
+    s"${id.value}_${f"${version.value}%05d"}.protocol"
+
   implicit def impl[F[_]: Sync](config: FileStorageConfig): Storage[F] =
```

Comment:
Why don't you just override the `apply` method instead of making an `implicit` method? On that way, you could use ` FileStorage` like `Resource.pure(FileStorage[F](fsc))`

Reply:
Yep, seems reasonable, just changed that.